app/mailers/step_mailer.rb

```diff
@@ -3,6 +3,15 @@ def step_reply_received(step)
     @proposal = step.proposal.decorate
     assign_threading_headers(@proposal)
     @step = step
+    last_approved_user = nil
+    
+    # Iterate through steps to find the LAST 'approved' person
```

Comment:
If the logic of the subsequent code is hard to parse, maybe put it in a well-named private method? Eg `last_step_user` (remember, these people are not always approvers! :smile: )